integration/testrunner/grpc.go

```diff
@@ -307,29 +353,45 @@ func createConnection(ctx context.Context, t *testing.T, addr, idToken string) *
 // This query is used to find the relevant audit log in BigQuery, which we assume will be added by the server.
 // We specifically look up the log using the UUID specified in the request as we know the server will add that
 // as the resource name, and provides us a unique key to find logs with.
-func (g *GRPC) makeQueryForGRPCUnary(id string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE jsonPayload.resource_name=?", g.ProjectID, g.DatasetQuery)
-	if g.RequireJustification {
-		// TODO(#265): For back-compatibility, we need to ensure justification in either metadata key.
-		queryString += ` AND jsonPayload.metadata.justification IS NOT NULL`
+func (g *GRPC) makeQueryForGRPCUnary(id string, fieldsNameMap [][]string) *bigquery.Query {
+	queryString := "SELECT "
+	for _, v := range fieldsNameMap {
+		queryString += fmt.Sprintf("%s as %s,", v[0], v[1])
```

Comment:
This is vulnerable to SQL injection

Reply:
Hi, I am not very familiar with this problem/issue here, is there any suggestions or a good reference for this?

Reply:
Updated.